addons/binding/org.openhab.binding.silvercrestwifisocket/src/main/java/org/openhab/binding/silvercrestwifisocket/internal/utils/WifiSocketPacketConverter.java

```diff
@@ -207,9 +212,21 @@ private SilvercrestWifiSocketResponse decryptResponsePacket(final String hexPack
                     + "]  \nDecryptedPacket:[" + decryptedData + "]");
         }
 
-        logger.trace("Decrypt success. Packet is from socket with mac address [{}] and type is [{}]", macAddress,
-                responseType);
-        return new SilvercrestWifiSocketResponse(macAddress, responseType);
+        SilvercrestWifiSocketVendor vendor = null;
+        for (SilvercrestWifiSocketVendor v : SilvercrestWifiSocketVendor.values()) {
```

Comment:
Maybe add this as static method to the `SilvercrestWifiSocketVendor`, it and its logic belong there.

Reply:
Will add a public static method fromCode to the Enum